libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -63,6 +51,33 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun requestFasterRoute(
+        adjustedRouteOptions: RouteOptions,
+        routesRequestCallback: RoutesRequestCallback
+    ) {
+        if (routes.isEmpty()) {
+            return
```

Comment:
We should call `routesRequestCallback#onRoutesRequestFailure` here.

Reply:
I think it should rather be `routesRequestCallback#onRoutesRequestCanceled` because failure means the request was made and failed for some reason; whereas in this case we don't even make the request. Wdyt?

Reply:
Sounds good 👌 @abhishek1508